homeassistant/components/plugwise/sensor.py

```diff
@@ -0,0 +1,346 @@
+"""Plugwise Sensor component for Home Assistant."""
+
+import logging
+
+from homeassistant.const import (
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_ILLUMINANCE,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_PRESSURE,
+    DEVICE_CLASS_TEMPERATURE,
+    PRESSURE_BAR,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.entity import Entity
+
+from . import SmileSensor
+from .const import (
+    COOL_ICON,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_VALVE,
+    DEVICE_STATE,
+    DOMAIN,
+    FLAME_ICON,
+    IDLE_ICON,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_TEMPERATURE = [
+    "Temperature",
+    TEMP_CELSIUS,
+    DEVICE_CLASS_TEMPERATURE,
+    "mdi:thermometer",
+]
+ATTR_BATTERY_LEVEL = ["Charge", "%", DEVICE_CLASS_BATTERY, "mdi:battery-high"]
+ATTR_ILLUMINANCE = [
+    "Illuminance",
+    "lm",
+    DEVICE_CLASS_ILLUMINANCE,
+    "mdi:lightbulb-on-outline",
+]
+ATTR_PRESSURE = ["Pressure", PRESSURE_BAR, DEVICE_CLASS_PRESSURE, "mdi:water"]
+SENSOR_MAP = {
+    "setpoint": ATTR_TEMPERATURE,
+    "temperature": ATTR_TEMPERATURE,
+    "intended_boiler_temperature": ATTR_TEMPERATURE,
+    "battery": ATTR_BATTERY_LEVEL,
+    "water_pressure": ATTR_PRESSURE,
+    "temperature_difference": ATTR_TEMPERATURE,
+    "electricity_consumed": [
+        "Current Consumed Power",
+        "W",
+        DEVICE_CLASS_POWER,
+        "mdi:flash",
+    ],
+    "electricity_produced": [
+        "Current Produced Power",
+        "W",
+        DEVICE_CLASS_POWER,
+        "mdi:flash",
+    ],
+    "electricity_consumed_interval": [
+        "Consumed Power Interval",
+        "Wh",
+        DEVICE_CLASS_POWER,
+        "mdi:flash",
+    ],
+    "electricity_produced_interval": [
+        "Produced Power Interval",
+        "Wh",
+        DEVICE_CLASS_POWER,
+        "mdi:flash",
+    ],
+    "outdoor_temperature": ATTR_TEMPERATURE,
+    "illuminance": ATTR_ILLUMINANCE,
+    "water_temperature": ATTR_TEMPERATURE,
```

Comment:
There is a lot going on here.  It looks like there are two types of sensor configurations.

Does it make sense to split the sensors up into two different SENSOR_MAP s?

Reply:
Yes, due to supporting many devices there can be a multitude of sensors.
We could divide them into three sensor_maps: temperature-, energy- and miscellaneous-type sensors. What do you think?

Reply:
Added the different sensor_maps as created by @bouwew